db/migrate/20170530102506_add_deleted_to_containers_tables.rb

```diff
@@ -0,0 +1,25 @@
+class AddDeletedToContainersTables < ActiveRecord::Migration[5.0]
+  def change
+    add_column :container_definitions, :deleted, :boolean, :default => false, :null => false
+    add_column :container_groups, :deleted, :boolean, :default => false, :null => false
+    add_column :container_images, :deleted, :boolean, :default => false, :null => false
+    add_column :container_projects, :deleted, :boolean, :default => false, :null => false
+    add_column :containers, :deleted, :boolean, :default => false, :null => false
```

Comment:
Not going to have this debate again. 😄   We intentionally avoid DB layer logic (foreign keys, default values, etc), so that we can maintain maximum flexibility for release branches.  That is, since we don't allow schema changes / db migrations on release branches, if we make a mistake we *can't* fix it.  (Additionally, when we were on rubyrep, these constraints had always proven to be pratically impossible for replication.  Now that we are on pglogical, that argument is probably moot.)

I've been hesitant to allow these through unless they are absolutely necessary (e.g. there'd have to be a compelling performance reason)